### PR TITLE
feat(batch): record fermentation gravity measurements and compute ABV (B2)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Mobile slice **B2** of the first-real-brew journey: fermentation gravity measurement entry. The brewer records the original gravity (OG) at the start of fermentation and the final gravity (FG) at the end; the app computes and **explains** ABV = (OG − FG) × 131.25 client-side. Backend is unchanged — the measurement API already existed.

- New **measurement entry screen** + route `batches/[id]/measurement`: OG/FG picker, numeric specific-gravity input, **client-side FG ≥ OG guard** with a plain-French explanation.
- **"Je n'ai pas de densimètre"** escape that never blocks the brew and shows buy-advice (no fabricated estimate); fermentation end is **taught, not auto-detected** (no `fermentation_completed_at` write).
- `BatchDetailsScreen` **"Densités & alcool"** card: the computed ABV + its explanation, or a guidance message when only OG (or no reading) exists.
- Clean Architecture: domain types, data-layer `createMeasurement` / `listMeasurements` over the shared http-client, application use-cases with the demo/live branch, plus `computeAbv` and `validateFinalGravity`.

## Context

P0 step toward a novice reaching bottling live (roadmap TRACKER → ASSISTANT). Decisions locked with the founder: SG format only, OG + FG only (interim readings deferred), teach/judge fermentation end, no-hydrometer escape, FG ≥ OG block, client-side ABV. Conception: `docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md`.

## Checklist

- [x] `ci:check` (lint + typecheck + format) green
- [x] Happy/Sad/Edge tests for every new unit (incl. a cache-invalidation regression); batches suite green
- [x] Conforms to the B2 conception + ADRs (no backend change, ABV client-side, no fermentation-completion write)
- [x] No `any`, named exports, theme tokens, no raw fetch, French UI copy

Refs the first real brew (B2 — US-0404).

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
Pour info : nouvelle étape du suivi d'un brassin sur téléphone — la saisie de la densité du moût.

Pendant la fermentation, le brasseur note la densité au départ (OG) puis à la fin (FG) avec un petit instrument appelé densimètre ; l'application calcule alors toute seule le taux d'alcool et explique d'où vient le chiffre. Pas de densimètre ? L'appli ne bloque rien et conseille d'en acheter un, sans inventer de valeur. Tout se passe côté téléphone : le serveur n'a pas changé.

Rien à faire de particulier : cette PR ajoute un écran et ses tests, sans toucher au reste.

File: packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+      void queryClient.invalidateQueries({
+        queryKey: ["batches", "measurements", batchId],
+      });
+      void queryClient.invalidateQueries({ queryKey: ["batches", "list"] });
+      router.back();
+    },
+  });
+
+  const parsedValue = parseFloat(valueText);
+  const hasValue = valueText.trim() !== "" && !Number.isNaN(parsedValue);
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx]: Fixed in 15cbceb — `parseFloat(valueText.replace(",", "."))` normalizes a FR comma before parsing, so "1,050" records as 1.050. Added a regression test ("1,050" → value 1.05). Good catch.

[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx]: Fixed in 15cbceb — the comma is normalized to a dot before parseFloat, so a FR-style "1,050" no longer collapses to 1 (which would have corrupted the saved OG/FG). Added an edge-case test.

File: packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+          densité finale (FG) à la fin. Saisis-la en densité spécifique, par
+          exemple 1.050.
+        </Text>
+
+        <View style={styles.kindRow}>
+          <Pressable
+            accessibilityRole="button"
+            style={[styles.kindTab, kind === "og" && styles.kindTabActive]}
+            onPress={() => handleSelectKind("og")}
+          >
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx:170]: Added `accessibilityState={{ selected: kind === 'og' }}` on the OG tab (15cbceb).

File: packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+              ]}
+            >
+              Densité initiale (OG)
+            </Text>
+          </Pressable>
+          <Pressable
+            accessibilityRole="button"
+            style={[styles.kindTab, kind === "fg" && styles.kindTabActive]}
+            onPress={() => handleSelectKind("fg")}
+          >
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx:185]: Added `accessibilityState={{ selected: kind === 'fg' }}` on the FG tab (15cbceb).

File: packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+        <Text style={styles.inputLabel}>{getKindLabel(kind)}</Text>
+        <TextInput
+          testID="measurement-value-input"
+          style={styles.textInput}
+          value={valueText}
+          onChangeText={setValueText}
+          keyboardType="numeric"
+          placeholder="1.050"
+        />
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/batches/presentation/MeasurementEntryScreen.tsx:205]: Changed to `keyboardType="decimal-pad"` (15cbceb) — better for specific-gravity entry, and the comma→dot normalization keeps parsing correct on either separator.

File: docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+sequenceDiagram
+  actor B as Brewer
+  participant M as Mobile (MeasurementEntryScreen)
+  participant UC as Use-case (recordBatchMeasurement)
+  participant API as Backend API (existing)
+  B->>M: Open my batch, tap "Saisir une densite"
+  M->>B: Form - pick OG or FG, enter SG (e.g. 1.050)
+  alt No hydrometer (escape, never a dead-end)
+    B->>M: Tap "Je n'ai pas de densimetre"
+    M-->>B: Buy-advice card; proceed without OG/FG; ABV marked unavailable (no estimate)
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md]: Fixed in 15cbceb — restored the accents in the diagram: "Saisir une densité", "Je n'ai pas de densimètre".

File: docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      alt value within SG range 0.99 to 1.2
+        API-->>UC: 201 MeasurementDto
+      else out of range
+        API-->>UC: 400 MeasurementValidationError
+        UC-->>M: surface the range error in plain words
+      end
+    end
+    UC-->>M: measurement saved
+    M->>M: invalidate batch cache, recompute ABV = (OG - FG) x 131.25
+    M-->>B: Show ABV + plain explanation (or "ABV calculee a la fin" if only OG yet)
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md]: Fixed in 15cbceb — "ABV calculée à la fin".

File: docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+- **OG + FG only for v1** (founder decision): interim `sg_spot` readings are **deferred** (atomic refinement once the app is used on real brews). Consequence captured below.
+- **Fermentation end = taught, not auto-detected** (founder decision): with OG/FG only the app cannot compute gravity stability, so it **teaches the rule** ("stable 2-3 days = done") and the brewer judges and records FG. The app does **not** auto-write `fermentation_completed_at`; completion stays an explicit user action on the existing independent lifecycle.
+- **No hydrometer = never a dead-end** (founder decision, US-0404): advise buying one, let the brew proceed, show **"alcool indisponible"** — **no** fabricated estimate (pedagogy).
+- **Implausible value guard** (founder decision): `FG >= OG` is blocked **client-side** with a plain explanation — this is distinct from the API's range `400` (which guards `[0.99, 1.2]`).
+- **ABV computed client-side** (KISS): `(OG - FG) x 131.25`, rounded to 1 decimal, in a pure helper; the `abv_actual` server snapshot is **deferred**. Pedagogy is the maitre-mot: the screen explains what OG/FG/ABV are (BeerXML/BJCP vocab: OG = densite initiale, FG = densite finale).
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply docs/architecture/diagrams/brew-day/02-sequence-record-gravity-measurement.md]: Fixed in 15cbceb — restored "maître-mot", "densité initiale", "densité finale" in the note.

**Comment by benoit-bremaud** [comment]
**End-of-review recap** — addressed the 8 Copilot + Codex comments in `15cbceb`:

- **Bug (comma decimal):** a FR-style `1,050` was parsed by `parseFloat` as `1`, silently corrupting the recorded gravity. Now normalized (`,`→`.`) before parsing, with a regression test.
- **Input:** `keyboardType="decimal-pad"` for specific-gravity entry.
- **A11y:** `accessibilityState={{ selected }}` on the OG/FG toggle tabs.
- **Docs:** restored the French accents in the B2 sequence diagram.

CI is re-running on the new commit; all review threads have a reply.
